lib/datasets.go

```diff
@@ -1013,7 +1013,7 @@ func (datasetImpl) Save(scope scope, p *SaveParams) (*dataset.Dataset, error) {
 		// apply the transform
 		shouldWait := true
 		transformer := transform.NewTransformer(scope.AppContext(), scope.Loader(), scope.Bus())
-		if err := transformer.Commit(scope.Context(), ds, runID, shouldWait, secrets); err != nil {
+		if err := transformer.Commit(scope.Context(), ref.InitID, ds, runID, shouldWait, secrets); err != nil {
```

Comment:
passing down the `ref.InitID`, rather than just populating `ds.ID` with the correct id. Based on the tests, it seems like we want this field to NOT be filled in when we go to save a dataset. If this is incorrect, let me know!

Reply:
It _should_ be fine, to set ds.Commit.ID within the save path. Frankly, it should be required. But that'll take some work. I'll file an issue for InitIDs in the save path, but this is fine for now